### PR TITLE
Cleaning up IPC edge case with packets larger than 4k

### PR DESCRIPTION
**Pull request by jongardiner**
Also, do a better job of detecting a crashed child and exiting the parent process.

### CODE REVIEW

File: src/SocketBuffer.php [hunk 1]
```diff
@@ -2,12 +2,27 @@
...
+		$index1 = strpos($str,"\n");
+		$index2 = strpos($str,"\r");
+		if ($index1 === false && $index2 === false) {
+			return false;
+		} else if ($index1 === false) {
+			return $index2;
+		} else if ($index2 === false) {
+			return $index1;
+		} else {
+			return min($index1, $index2);
```
Comment (jlesueur) [inline src/SocketBuffer.php]: if you get "\r\n", I think this means you'll get an empty string between them as a message? But I'm guessing the children aren't commonly producing windows line endings. Will they if run in windows?

File: src/SocketBuffer.php [hunk 2]
```diff
@@ -16,17 +31,15 @@ function read($socket) {
...
-					}
-					$last = $index + 1;
+			do {
+				$index = self::firstEol($this->buffer);
+				if ($index !== false) {
+					$this->messages[] = trim(substr($this->buffer, 0, $index + 1), "\r\n\0\x0B");
+					$this->buffer = substr($this->buffer, $index + 1);
 				}
-			}
-			$this->buffer = substr($this->buffer, $last + 1);
```
Comment (jlesueur) [inline src/SocketBuffer.php]: if there was a message > 4k, then this line would strip off one character from the front before it grabbed the next 4k block. Have I grasped the original bug?

Comment (jongardiner) [reply src/SocketBuffer.php]: Yes.  And as it happened, the last message that includes the timings for all the plugins just recently crossed the 4k boundary, so it was the first time we had exercised that code.

File: src/SocketBuffer.php (hunk 1 above)
Comment (jongardiner) [reply src/SocketBuffer.php]: Those could happen here, but in the parent function we ignore empty messages.

**Pull request merged**